### ISSUE TITLE
[ALM Accelerator - QUESTION] What Permissions are Needed to Sign In the CustomAzureDevOps Connection

### ISSUE THREAD

**Issue by jfung85** [microsoft/coe-starter-kit#5548]
### What is your question?

We have configured the ALM Accelerator solution and are trying to provide access to a different account.

When logging into the app, we tried following this article: https://learn.microsoft.com/en-us/power-platform/guidance/alm-accelerator/new-maker-experience?source=recommendations

We are specifically having an issue with the CustomAzureDevOps connection. 
![image](https://github.com/microsoft/coe-starter-kit/assets/97466777/6a23787e-66f5-4ccb-b537-af4ba31dc74b)

We have added the user to the environment with Basic User/Service Reader/Service Writer. But we receive this error:
The user does not have sufficient access rights to perform this operation on entity with id  in instance with URL 'instanceurl' . Required access right 'ReadAccess' is missing.

### What component are you experiencing the issue with?

ALM Accelerator Canvas App

### What solution version are you using?

Accelerator Version: 1.0.20230405.2(Managed)

**Comment by jfung85** [comment]
We found this configuration was the resolution: Go to the ALM Accelerator solution -> Custom connectors -> Click on CustomAzureDevOps -> Click on Share -> Add user with Can view permissions

**Comment by jfung85** [comment]
After getting past the ReadAccess error, the next error below came up on login. I think this still might fit within the frame of the question. Looking for articles on how to configure this on an App Registration but haven't found any direct guidance yet. 

AADSTS50105: Your administrator has configured the application ALMAcceleratorServicePrincipal to block users unless they are specifically granted ('assigned') access to the application. The signed in user 'username' is blocked because they are not a direct member of a group with access, nor had access directly assigned by an administrator. Please contact your administrator to assign access to this application.

**Comment by mikefactorial** [comment]
@jfung85 good to know you were able to get past the first issue. Seems like both of these are related. Generally, you want to manage user access using AAD Groups and Group Teams in Dataverse to make the process of on-boarding new users easier. If you share the connector with a group team in Dataverse then they will get access when they are added to the group in AAD. Similarly sharing the app with the group team means users will also get access to the app and by extension sharing the process records in Dataverse associated with the flows the app uses with a group team will ensure they have access to everything they need by virtue of adding them to the AAD Group. Additionally, there are 2 security roles that ship with the ALM Accelerator (ALM Power App Access and ALM Power App Deployment Configuration). You can assign those roles to the group team to provide either access to the app, in the case of the former, and / or access to the model driven admin app, in the case of the latter. Will work on updating the docs to be a little more explicit about what I've explained above in this doc. https://learn.microsoft.com/en-us/power-platform/guidance/alm-accelerator/setup-app-user-permissions#dataverse-user-setup

**Comment by mikefactorial** [comment]
NOTE: You can also use AAD Groups in Azure DevOps in a similar manner to make adding / removing users simpler.

**Comment by jfung85** [comment]
Hi @mikefactorial, thanks for the AAD group tip. We were configuring one account for testing out the configuration but I agree a group would be easier to manage for a long term configuration. Here are the changes I made from your comments and article but I'm still seeing the same AADSTS50105 error. Am I missing anything else?
- Custom Connector -> shared to user account with Can view permissions (note: tried switching to a group but original error came up, not sure if it's a propagation issue, but switched back to individual user for sake of testing)
- Environment Group Team Role and Specific User Account -> Basic User and ALM Power App Access 
- Azure Dev Op -> <Solution Project> -> Project Settings -> Service Connections -> Security -> Added user account with "User" role (Notes (1) System added the user but switched Role to Creator and Access to Inherited (2) I tried to add an AAD group here but after clicking Add, nothing changed on the listing)

**Comment by mikefactorial** [comment]
@jfung85 after reviewing the error message again it looks as if the issue isn't a problem with Dataverse or AzDO permissions but rather an issue with permissions to the Service Principal (AAD App Registration). Based on the error the App Registration was setup to require explicit permissions to use the service principal configured on the custom connector. If this was setup by you, you may want to follow the steps in this doc to add the user https://learn.microsoft.com/en-us/azure/active-directory/manage-apps/add-application-portal-assign-users

**Comment by jfung85** [comment]
We were able to resolve this by configuring the user permission on the Enterprise Application. One key point is that this is different from the App Registration although they may have the same name. And it was a little difficult to find through the search interface in Azure Portal. We needed to remove the filter and search by Application ID which can be found on the App Registration. Once we found it, we needed to have an owner account add the test user account and provided the Default Access role. This was the final piece that allowed a different user to open the ALM Accelerator for Power Power Platform app.
